ERCS/erc-7828.md

```diff
@@ -14,227 +14,208 @@ requires: 155, 7930
-    2. Call `resolver(namehash(8000000a.reverse))` on the mainnet ENS registry. It'll return `0x00000beef055f7934784d6d81b6bc86665630dba`, the address of the `L2ReverseRegistrar` *on the network with chainid 10*.
-8.  Verify the return of the `coinType` method of the `L2ReverseRegistrar` matches the coinType as serialized in step 7.
-9.  Call `nameForAddr(bytes address)` with the address as serialized in step 8. This will return the human-readable name `alice.eth`.
-10.  Check forward resolution of the name as described in ENSIP-11 and repeated in the section above.
-    - If it resolves to the same address, then proceed normally.
-    - If forward resolution of the ENS name returns an address different from the original, or returns an empty byte array, the wallet MUST display the raw, human-readable [ERC-7930] serialized address, indicating the address is unresolved. The wallet SHOULD also warn the user of the mismatch to prevent potential spoofing or confusion.
-11.  Compute the checksum as described in [ERC-7930]: `0xC69BEB13`.
-12.  Format the address in the format `<address>@<chain>#<checksum>` by removing `l2.eth` from step 3 and display it to the user: `alice.eth@optimism#C69BEB13`
+This section is non-normative, as future ENSIPs can modify the way in which ENS text records, and arbitrary data are resolved using ENS. Its purpose is to demonstrate how a human-readable _Interoperable Name_ is resolved to a binary [ERC-7930] _Interoperable Address_ that can be used for protocol level interactions using current ENSIP standards.
+
+The implementation of `ChainResolver` is an all-in-one registry-resolver assigned as the resolver on `<namespace.eth>`. The implementation supports all current ENSIPs relating to resolution, through its implementation of the `IExtendedResolver` interface defined in ENSIP-10: Wildcard Resolution.
+
+### Forward resolution
+
+The example below describes a possible flow for a user facing wallet. Lets assume that the user wants to send funds to a user on Optimism.
+
+#### Step 1: User Input
+1. Let the user input an ENS name. For examples, `alice.eth`.
+2. Let the user select a chain. In this case, the user would select 'Optimism'.
+
+#### Step 2: Build the _Interoperable Name_
+3. The underlying label for 'Optimism' is 'optimism'. Append `<namespace.eth>` to that label => `optimism.<namespace>.eth`
+
+#### Step 3: Resolve the [ERC-7930] _Interoperable Address_
+4. Discern the resolver for `optimism.<namespace>.eth` and resolve `data(bytes32 namehash, string key)` (ENSIP-24: Arbitrary Data Resolution) passing in the `namehash` (see ENSIP-1) of `optimism.<namespace>.eth` and the key, `chain-id`. 
+
+#### Step 4: Resolve alice.eth **for** the specified chain
+5. Extract the CAIP-2 `ChainReference` from the [ERC-7930] identifier returned in the previous step.
+6. Convert the `ChainReference` into an ENS `coinType` (see ENSIP-11).
```

Comment:
For non-EVM chains, ENSIP-11 does not apply. We reference SLIP-44 for chain identification.